onnxruntime/python/tools/kernel_explorer/kernels/kernel_explorer.py

```diff
@@ -60,3 +60,86 @@
 
 # pylint: disable=wrong-import-position, disable=unused-import, disable=wildcard-import
 from _kernel_explorer import *  # noqa
+
+"""Benchmark Reporter"""
+
+from abc import abstractmethod
+from contextlib import contextmanager
+from dataclasses import dataclass
+
+
+@dataclass
+class InstanceStatus:
+    name: str
+    dtype: str
+    milliseconds_duration: float
+
+    def __lt__(self, other):
+        if "Tunable" in self.name or other.duration < 0:
+            return True
+        if "Tunable" in other.name or self.duration < 0:
+            return False
+
+        return self.duration < other.duration
+
+    @property
+    def duration(self):
+        return self.milliseconds_duration * 1000
+
+    @abstractmethod
+    def report(self) -> str:
+        ...
+
+
+@dataclass
+class ComputeStatus(InstanceStatus):
+    FLOPs: int
+
+    @property
+    def tflops(self):
+        return self.FLOPs * 1e6 / self.duration / 1e12
+
+
+@dataclass
+class BandwidthStatus(InstanceStatus):
+    bytes: int
+
+    @property
+    def gbps(self):
+        return self.bytes * 1e6 / self.duration / 1e9
+
+
+class InstanceBenchmarkReporter:
+    def __init__(self):
+        self.sort = False
+        self.reporters = []
+
+    def set_sort(self, sort):
+        self.sort = sort
+
+    def flush(self):
+        self.reporters.sort()
+        for item in self.reporters:
+            print(item.report())
+        self.reporters.clear()
+
+    def receive(self, status):
+        self.reporters.append(status)
+        if not self.sort:
+            self.flush()
```

Comment:
> if not self.sort:
            self.flush()
            
Why do we call `flush()` immediately when no sorting is requested? I would not expect the behavior (i.e., timing of output in this case) of a reporter class to change depending on whether a sort was requested or not. Can we fix this to ensure uniform behavior?

Reply:
Then we need a better name, instead of `flush`. Something like `make_report`

Reply:
That's because for some op like Gemm, CK will provide nearly 100 instances and it will take a long time to run all the instances. we need this for quick check instead of waiting for all instances done.